test/SDTest.cpp

```diff
@@ -20,6 +21,30 @@ unittest(singleton) {
   assertEqual(thing1, thing2);
 }
 
+unittest(tankControllerLoop) {
+  char data[250];
+  TankControllerLib* tc = TankControllerLib::instance();
+  DateTime_TC d1(2021, 4, 15);
+  d1.setAsCurrent();
+  assertFalse(SD.exists("20210415.csv"));
+  tc->loop();
+  delay(1000);
+  tc->loop();
+  assertTrue(SD.exists("20210415.csv"));
+  File file = SD.open("20210415.csv");
+  assertTrue(file.size() < sizeof(data));
+  if (file.size() < sizeof(data)) {
+    file.read(data, file.size());
+    data[file.size()] = '\0';
+    assertEqual(
+        "time,tankid,temp,temp setpoint,pH,pH setpoint,onTime,Kp,Ki,Kd\n"
+        "04/15/2021 00:00:00,   0, -242.02, 20.00, 0.000, 8.100,    1, 100000.0,      0.0,      0.0\n"
+        "04/15/2021 00:00:01,   0, -242.02, 20.00, 0.000, 8.100,    2, 100000.0,      0.0,      0.0\n",
+        data);
+  }
+  file.close();
+}
+
```

Comment:
Now we need a test similar to the one above that replaces the temperature and pH with 'C' during calibration.